### PR TITLE
[CI] Add options for release test run to run only likely failing tests

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?

In https://docs.google.com/document/d/1JC_dHCBkmU7dZY6CaqCFI55zRQo-zNnBeDq5BUGwdBg/edit#heading=h.kbv9vy7pjqaz, I made a proposal to using a test-behavior based coverage to run only tests that are likely to fail. At the same time, we would also want to run tests more often, so bisecting and creating issues for failing test is more prompt.

This PR add flags to build-pipeline.py to be able to run only tests that are likely to fail. Also add option to run tests that belong to a certain team.

## Related issue number

Issue #34242 

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Release tests - https://buildkite.com/ray-project/release-tests-pr/builds/34595

### CODE REVIEW

File: release/ray_release/scripts/build_pipeline.py [hunk 1]
```diff
@@ -34,6 +34,19 @@
     type=str,
     help="File containing test configurations",
 )
+@click.option(
+    "--only-likely-failing-tests",
```
Comment (aslonnie) [inline release/ray_release/scripts/build_pipeline.py:38]: maybe rename to `--likely-failing-tests-only`? or just `--likely-failing-only`?

I tried to think about a single word to replace "likely failing", but my English failed me.. so I asked chatgpt, and it gives me "precarious".

so maybe `--precarious-only` or `--precarious-tests-only`?

**Review by aslonnie — APPROVED** [review]
looks good in general. just flag naming.

File: release/ray_release/scripts/build_pipeline.py (hunk 1 above)
Comment (can-anyscale) [reply release/ray_release/scripts/build_pipeline.py:38]: Nice ;))

Comment (aslonnie) [reply release/ray_release/scripts/build_pipeline.py:38]: and it also gives me 3 more options for backup:

image

Comment (can-anyscale) [reply release/ray_release/scripts/build_pipeline.py:38]: lol i like 'tenuous', sound like an octopus

File: release/ray_release/buildkite/filter.py [hunk 2]
```diff
@@ -16,14 +18,37 @@ def _unflattened_lookup(lookup: Dict, flat_key: str, delimiter: str = "/") -> An
...
+    s3 = boto3.client("s3")
+    files = s3.list_objects_v2(
+        Bucket="ray-release-automation-results",
+        Prefix="continuous-release/",
+    )["Contents"]
+    latest_coverage = [file for file in sorted(files, key=_get_last_modified)][-1]
+    data = s3.get_object(
+        Bucket="ray-release-automation-results", Key=latest_coverage["Key"]
+    )
+    return json.loads(data["Body"].read().decode("utf-8"))
```
Comment (krfricke) [inline release/ray_release/buildkite/filter.py:34]: Where do we write this file? Can we maybe add a few comments how this file looks like and how the results look like?

Lastly, can we maybe modularize this a bit. E.g. have a utility in `aws.py` to fetch a file from S3, and then call the utility here? It also seems to me like we should have at least constants defined for the bucket + file location

**Review by krfricke — CHANGES_REQUESTED** [review]
Left some suggestions. Ping me for merge

**Pull request closed without merging**